src/App.ts

```diff
@@ -407,29 +407,39 @@ export default class App {
     // Set body and payload (this value will eventually conform to AnyMiddlewareArgs)
     // NOTE: the following doesn't work because... distributive?
     // const listenerArgs: Partial<AnyMiddlewareArgs> = {
-    const listenerArgs:
-      Pick<AnyMiddlewareArgs, 'body' | 'payload'> & {
-        /** Say function might be set below */
-        say?: SayFn
-        /** Respond function might be set below */
-        respond?: RespondFn,
-        /** Ack function might be set below */
-        ack?: AckFn<any>,
-      } = {
-        body: bodyArg,
-        payload:
-          (type === IncomingEventType.Event) ?
-            (bodyArg as SlackEventMiddlewareArgs['body']).event :
-          (type === IncomingEventType.ViewAction) ?
-            (bodyArg as SlackViewMiddlewareArgs['body']).view :
-          (type === IncomingEventType.Action &&
-            isBlockActionOrInteractiveMessageBody(bodyArg as SlackActionMiddlewareArgs['body'])) ?
-            (bodyArg as SlackActionMiddlewareArgs<BlockAction | InteractiveMessage>['body']).actions[0] :
-          (bodyArg as (
-            Exclude<AnyMiddlewareArgs, SlackEventMiddlewareArgs | SlackActionMiddlewareArgs | SlackViewMiddlewareArgs> |
-            SlackActionMiddlewareArgs<Exclude<SlackAction, BlockAction | InteractiveMessage>>
-          )['body']),
-      };
+    const listenerArgs: Pick<AnyMiddlewareArgs, 'body' | 'payload'> & {
+      /** Say function might be set below */
+      say?: SayFn
+      /** Respond function might be set below */
+      respond?: RespondFn,
+      /** Ack function might be set below */
+      ack?: AckFn<any>,
+    } = {
+      body: bodyArg,
+      payload:
+        type === IncomingEventType.Event
+          ? (bodyArg as SlackEventMiddlewareArgs['body']).event
+          : type === IncomingEventType.ViewAction
+            ? (bodyArg as SlackViewMiddlewareArgs['body']).view
+            : type === IncomingEventType.Action &&
+              isBlockActionOrInteractiveMessageBody(
+                bodyArg as SlackActionMiddlewareArgs['body'],
+              )
+              ? (bodyArg as SlackActionMiddlewareArgs<
+                BlockAction | InteractiveMessage
+              >['body']).actions[0]
+              : (bodyArg as (
+                | Exclude<
+                  AnyMiddlewareArgs,
+                  | SlackEventMiddlewareArgs
+                  | SlackActionMiddlewareArgs
+                  | SlackViewMiddlewareArgs
+                >
+                | SlackActionMiddlewareArgs<
+                  Exclude<SlackAction, BlockAction | InteractiveMessage>
+                >
+              )['body']),
+    };
```

Comment:
prettier did some auto-formatting here, seems better maybe? happy to change back.

Reply:
i personally prefer the alignment of the "cases" under the payload property in the original.

i don't mind moving `Pick<AnyMiddlewareArgs, 'body' | 'payload'> & {` up a line. would it be cook to make that change without all the other stuff?

Reply:
done 👍